### PR TITLE
DRS Localizer Azure token strategy implementation [BT-436]

### PR DESCRIPTION
**Pull request by mcovarr**
Tested using the included DRS Localizer repo template. Follow Janet's instructions for getting a B2C token, setting up a Key Vault, and storing the former in a secret within the latter. You should then be able to run the repo template with the coordinates for vault name, secret name, and client id plugged in to the invocation. Assuming your token is fresh (mine expired several times in one day while testing this), everything should Just Work.

Oh and `getm` must be installed:

```
pip3 install getm==0.0.4
```

### CODE REVIEW

**Review by jgainerdewar — APPROVED** [review]
LGTM, I tested using the nice new repo template and it downloaded my file 👍 👍

File: cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/GetmChecksum.scala [hunk 1]
```diff
@@ -1,35 +1,57 @@
...
+
+case class Md5(override val rawValue: String) extends GetmChecksum {
+  override def value: ErrorOr[String] = {
+    val trimmed = rawValue.trim
+    if (trimmed.matches("[A-Fa-f0-9]+"))
+      trimmed.validNel
+    // Azure reports its md5s in base64, but `getm` knows only hex. For the sanity of humans storing data in Azure
+    // it's probably best if the DRS localizer does this conversion for `getm`.
+    else if (isBase64(trimmed))
+      (trimmed |> decodeBase64 |> encodeHexString).validNel
```
Comment (breilly2) [inline cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/GetmChecksum.scala:36]: I'd rather we not have this in the code, but that depends on https://broadworkbench.atlassian.net/browse/DR-2259 getting done. I'm thinking of adding that link to the comment above and creating a ticket on our board to remove this condition. Thoughts?

File: project/Dependencies.scala [hunk 2]
```diff
@@ -516,16 +516,16 @@ object Dependencies {
 
   val bcsBackendDependencies: List[ModuleID] = commonDependencies ++ refinedTypeDependenciesList ++ aliyunBatchComputeDependencies
 
-  val tesBackendAzureDependencies: List[ModuleID] = List(
+  val azureDependencies: List[ModuleID] = List(
```
Comment (breilly2) [inline project/Dependencies.scala:519]: I thought the reason for having "tes" in the name is that some of these dependencies are needed for the TES backend. Need to go back through git history.

**Review by breilly2 — COMMENTED** [review]
I think this is mostly good. Leaving initial comments, and I'm thinking of doing some sort of manual integration test myself before approving.

File: cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/GetmChecksum.scala (hunk 1 above)
Comment (jgainerdewar) [reply cromwell-drs-localizer/src/main/scala/drs/localizer/downloaders/GetmChecksum.scala:36]: Sounds like a good plan! I agree that we should remove this eventually but don't want it to block this merge.

File: project/Dependencies.scala (hunk 2 above)
Comment (jgainerdewar) [reply project/Dependencies.scala:519]: Previously these dependencies were used exclusively in the TES backend, now they're being used in the standalone DRS localizer as well.

File: build.sbt [hunk 3]
```diff
@@ -385,7 +385,7 @@ lazy val perf = project
 lazy val `cromwell-drs-localizer` = project
   .withExecutableSettings("cromwell-drs-localizer", drsLocalizerDependencies, drsLocalizerSettings)
   .dependsOn(`cloud-nio-impl-drs`)
-  .dependsOn(common % "test->test")
+  .dependsOn(common)
```
Comment (breilly2) [inline build.sbt:388]: Any idea what this change is about? Seems unrelated to the rest of this PR.

Comment (jgainerdewar) [reply build.sbt:388]: I had no idea what this was but found this SO thread: https://stackoverflow.com/questions/8193904/sbt-test-dependencies-in-multiprojects-make-the-test-code-available-to-dependen

Looks like this change removes a dependency on the test code in `common`, I assume because it wasn't actually being used.

File: project/Dependencies.scala (hunk 2 above)
Comment (breilly2) [reply project/Dependencies.scala:519]: But in just the past couple of days, this has gone from being called `tesBackendDependencies` to `azureDependencies` and I'm afraid we're losing context with that change. Maybe we should split these out into a couple sublists and join them together as needed depending on the artifact being built.

**Review by breilly2 — COMMENTED** [review]
A couple more comments. It seems Github's "post single comment" isn't working for me right now. :(

File: project/Dependencies.scala (hunk 2 above)
Comment (jgainerdewar) [reply project/Dependencies.scala:519]: Ah - during review in my recent PR that added this, there was a desire to handle these in a separate list and call them out as Azure-specific and not general TES dependencies. Note that this list we're talking about wasn't ever called `tesBackendDependencies`, rather these libraries used to be part of the existing `tesBackendDependencies` list, rather than a separate list that is included in `tesBackendDependencies`.

I don't know if that clarifies anything or further confuses things, happy to chat about this off-github if that's easier. I do think that the current state is a good representation of how the libraries are being used.

Comment (breilly2) [reply project/Dependencies.scala:519]: Yeah... it would help if I read the code correctly. 🙄  Thank you for clarifying.

**Review by breilly2 — APPROVED** [review]
👍   Okay, I'm happy with this code. I didn't end up testing myself because I saw that Janet tested and I trust her results.
